tests/AccountTests.test.ts

```diff
@@ -0,0 +1,241 @@
+        expect(tryLogin.id).not.toEqual(null);
+    });
+
+    it('login with no matching user', async () => {
+        await expect(() =>
+            AccountController.tryLogin(
+                'vanforeest@notexisting.de',
+                'verySecurePW'
+            )
+        ).rejects.toThrow(
+            new LoernwerkError(
+                'Mail/Name not matching an existing User',
+                LoernwerkErrorCodes.NOT_FOUND
+            )
+        );
+    });
+    it('login with wrong password', async () => {
+        const mockUser = await DBUser.findBy({ name: 'magnus' });
+        expect(mockUser.length).toEqual(1);
+        await expect(() =>
+            AccountController.tryLogin(mockUser[0].name, 'wrongPassword')
+        ).rejects.toThrow(
+            new LoernwerkError(
+                'Incorrect Password',
+                LoernwerkErrorCodes.NOT_FOUND
+            )
+        );
+    });
+
+    //getAccountById function
+    it('get existing account', async () => {
+        const mockUser = await DBUser.findBy({ name: 'magnus' });
+        expect(mockUser.length).toEqual(1);
+        const returnedUser = await AccountController.getAccountById(
+            mockUser[0].id
+        );
+        const hashComparison = await bcrypt.compare(
+            'verySecurePW',
+            returnedUser.password
+        );
+        expect(returnedUser).toBeInstanceOf(DBUser);
+        expect(returnedUser.name).toEqual('magnus');
+        expect(hashComparison).toEqual(true);
+        expect(returnedUser.mail).toEqual('magnus@carlsen.de');
+        expect(returnedUser.id).not.toEqual(null);
+    });
+
+    it('try getting nonexistent account', async () => {
+        await expect(() =>
+            AccountController.getAccountById(correctUser.id)
+        ).rejects.toThrow(
+            new LoernwerkError(
+                'No existing User with given ID',
+                LoernwerkErrorCodes.NOT_FOUND
+            )
+        );
+    });
+
+    //getAllAccounts function
+    it('get all accounts', async () => {
+        const allAccounts = await AccountController.getAllAccounts();
+        expect(allAccounts[0]).toBeInstanceOf(DBUser);
+        expect(allAccounts[0].name).not.toBe(null);
+        expect(allAccounts[0].id).not.toBe(null);
+    });
+
+    //deleteAccount function
+    it('delete a correct account', async () => {
+        const accountToBeDeleted = await DBUser.findOne({
+            where: { id: 931943 },
+        });
+        await AccountController.deleteAccount(accountToBeDeleted.id);
+        await expect(() =>
+            SequenceController.getSequenceByCode('CODE66')
+        ).rejects.toThrow(
+            new LoernwerkError(
+                'No matching sequence',
+                LoernwerkErrorCodes.NOT_FOUND
+            )
+        );
+    });
+
+    /**
+     * ensureAdminExists function
+     *NOTE: Admin email returns undefined@loernwerk.de, intended?
```

Comment:
that is already fixed

Reply:
Even with latest version of main merged, undefined@loernwerk.de is still returned